src/index.tsx

```diff
@@ -1,100 +1,114 @@
-import { Hono } from 'hono';
-import { neon } from '@neondatabase/serverless';
-import { drizzle } from 'drizzle-orm/neon-http';
-import { runners } from './db/schema';
-import { Resend } from 'resend';
-import { EmailTemplate } from './email_template';
 import { instrument } from "@fiberplane/hono-otel";
+import { neon } from "@neondatabase/serverless";
+import { drizzle } from "drizzle-orm/neon-http";
+import { Hono } from "hono";
+import { Resend } from "resend";
+import { EmailTemplate } from "./EmailTemplate";
+import { runners } from "./db/schema";
 
 type Bindings = {
   DATABASE_URL: string;
   RESEND_API: string;
 };
 
-const app = new Hono<{ Bindings: Bindings }>()
-
-app.get('/', (c) => {
-  return c.text('Hello Hono!')
-})
+const app = new Hono<{ Bindings: Bindings }>();
 
+app.get("/", (c) => {
+  return c.text("Hello Hono!");
+});
 
 enum Distance {
-  TEN = '10',
-  HALF = 'half',
-  FULL = 'full',
+  TEN = "10",
+  HALF = "half",
+  FULL = "full",
 }
 
+app.post("/api/marathon-sequential", async (c) => {
+  const { firstName, lastName, email, address, distance } = await c.req.json();
 
-app.post('/api/marathon-sequential',async (c) => {
- 
-  const { firstName, lastName, email, address, distance } = await c.req.json()
-
-  await insertData(firstName, lastName, email, address, distance, c.env.DATABASE_URL);
-  await sendMail(email, c.env.RESEND_API, firstName)
-
-  return c.text('Thanks for registering for our Marathon', 200)
-
-})
-
-app.post('/api/marathon-fire-and-forget', async (c) => {
-  const { firstName, lastName, email, address, distance } = await c.req.json()
-  
-  await insertData(firstName, lastName, email, address, distance, c.env.DATABASE_URL).catch(console.error)
-  
-  sendMail(email, c.env.RESEND_API, firstName).catch(console.error)
-
-  
-    return c.text('Thanks for registering for our Marathon', 201)
+  await insertData(
+    firstName,
+    lastName,
+    email,
+    address,
+    distance,
+    c.env.DATABASE_URL,
+  );
```

Comment:
so you'll notice i removed `.catch(console.error)` -- the reason i did this is that `catch` will actually handle the error, and you won't end up throwing an exception. 

in this scenario, we actually want to throw an error if the database insert fails, otherwise the code will continue to the final `return` statement, and the participant would see the "Thank you..." message even if the db insert failed